src/openloco/windows/construction/Construction.h

```diff
@@ -0,0 +1,523 @@
+            widget_end(),
+        };
+
+        static window_event_list events;
+        constexpr uint64_t enabledWidgets = common::enabledWidgets | allConstruction;
+        void tabReset(window* self);
+        void init_events();        
+        void drawTrack(uint16_t x, uint16_t y, uint32_t edi, uint8_t bh, uint32_t edx);
+        void drawRoad(uint16_t x, uint16_t y, uint32_t edi, uint8_t bh, uint32_t edx);
+    }
+
+    namespace station
+    {
+        enum widx
+        {
+            station = 8,
+            station_dropdown,
+            image,
+            rotate,
+        };
+
+        static widget_t widgets[] = {
+            commonWidgets(138, 190, string_ids::stringid_2),
+            make_widget({ 3, 45 }, { 132, 12 }, widget_type::wt_18, 1, 0xFFFFFFFF, string_ids::tooltip_select_station_type),
+            make_widget({ 123, 46 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown, string_ids::tooltip_select_station_type),
+            make_widget({ 35, 60 }, { 68, 68 }, widget_type::wt_3, 1),
+            make_widget({ 112, 104 }, { 24, 24 }, widget_type::wt_9, 1, image_ids::rotate_object, string_ids::rotate_90),
+            widget_end(),
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << station) | (1 << station_dropdown) | (1 << image) | (1 << rotate);
+
+        static window_event_list events;        
+        void tabReset(window* self);
+        void init_events();
+    }
+
+    namespace signal
+    {
+        enum widx
+        {
+            signal = 8,
+            signal_dropdown,
+            both_directions,
+            single_direction,
+        };
+        
+        static widget_t widgets[] = {
+            commonWidgets(138, 167, string_ids::stringid_2),
+            make_widget({ 3, 45 }, { 132, 12 }, widget_type::wt_18, 1, 0xFFFFFFFF, string_ids::tooltip_select_signal_type),
+            make_widget({ 123, 46 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown, string_ids::tooltip_select_signal_type),
+            make_widget({ 27, 110 }, { 40, 40 }, widget_type::wt_9, 1, 0xFFFFFFFF, string_ids::tooltip_signal_both_directions),
+            make_widget({ 71, 110 }, { 40, 40 }, widget_type::wt_9, 1, 0xFFFFFFFF, string_ids::tooltip_signal_single_direction),
+            widget_end(),
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << signal) | (1 << signal_dropdown) | (1 << both_directions) | (1 << single_direction);
+
+        static window_event_list events;        
+        void tabReset(window* self);
+        void init_events();
+    }
+
+    namespace overhead
+    {
+        enum widx
+        {
+            checkbox_1 = 8,
+            checkbox_2,
+            checkbox_3,
+            checkbox_4,
+            image,
+            track,
+            track_dropdown,
+        };
+
+        static widget_t widgets[] = {
```

Comment:
You will need to move these into the individual .cpp files and have an `extern widget_t widgets[{size}];`